packages/alfa-media/src/media.ts

```diff
@@ -232,20 +202,213 @@ export namespace Media {
       value: Value.JSON | null;
     }
 
-    export type Value = Number | String | Length | Angle | Percentage;
+    export function tryFrom(name: string): Result<Feature, string>;
+
+    export function tryFrom<T>(
+      name: string,
+      value: Value<T>
+    ): Result<Feature, string>;
+
+    export function tryFrom(
+      name: string,
+      value?: Value
+    ): Result<Feature, string> {
+      if (value) {
+        switch (name) {
+          case "width":
+            if (value.hasValue(Length.isLength)) {
```

Comment:
I'd rather keep the validations inside the classes themselves for better locality.
Essentially, it's `Width` who needs to know what it accepts as values. `tryFrom` shouldn't need to know what each of the individual feature accepts but simply ask them to validate the value.

Reply:
That's fine by me as well; I'll branch out `Feature.tryFrom()` into the different specific features. That's just good ol' divide and conquer 😄